scheduler/node/node.go

```diff
@@ -0,0 +1,86 @@
+package node
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/docker/swarm/cluster"
+)
+
+// Node is an abstract type used by the scheduler
+type Node struct {
+	ID         string
+	IP         string
+	Addr       string
+	Name       string
+	Cpus       int64
+	Memory     int64
+	Labels     map[string]string
+	Containers []*cluster.Container
+	Images     []*cluster.Image
+
+	UsedMemory  int64
+	UsedCpus    int64
+	TotalMemory int64
+	TotalCpus   int64
+
+	IsHealthy bool
+}
+
+// NewNode creates a node from an engine
+func NewNode(e *cluster.Engine) *Node {
```

Comment:
It was in the engine at the beginning, but I moved it here because, for now (it might change) we are also doing engine -> node in api/api.go (in `proxyRandom)